app/javascript/controllers/transcript_controller.js

```diff
@@ -26,17 +26,13 @@ export default class extends Controller {
     const tracks = this.player.textTracks_?.tracks_
     if (!tracks) return []
 
-    return tracks.filter(track => track.kind === 'captions')
+    return tracks.filter(track => track.kind === 'captions' && track?.cues?.cues_)
   }
 
   get cuesByLanguage() {
     const cues = {}
-    if (this.captionTracks.length == 0) return cues
```

Comment:
`this.captionTracks` always returns an array, and we can safely `forEach` an empty array below, so I don't see a need for this check.